### 02-09-19 - Aron, Alex, Karl
blastp kirc into genbank, blastp now has taxa. Selection by relevance, if there is already studies of a crypton in a genome we can sole this one out. Take a crypton from each crypton group to start of with. There will probably be a better assembly and more hits of crypton_F sequences, due to better annotation. 

* Make sure now to get acquianted with blastp and maybe the flanking sequence script on the lab site. 
* Get access to rackham
* Find paper or method on searching for unannotated sequences in eukaryotic genomes

### 16-09-19 - Aron, Alex, Karl
Scripts that can extract top hits for alignments. Use shorten scaffold script, not rename, to get the corrected genome. See it to be done with the sequences by tomorrow 17-09-19 16:30. JC on monday (23-09-19) monday 11:00.

### 09-10-19 
to reverse sequences in bioedit you can put \_R\_ in front of the sequence to put it in reverse. Keep track of the cryptons and the notes on crypton in a spread-sheet.

* Do they have complete edges of the transposons
* Use advanced consensus maker: https://www.hiv.lanl.gov/content/sequence/CONSENSUS/AdvCon.html to create consensus of homolugous seuqneces. 
* What species was it found in? 
* Important to keep notes of the annotative steps.

### 09-10-19
keep the unused sequences in a separate folder, use the spread sheet to annotate how many unsused ones there were. 

Seems that the results are comming back to be same same hits in the genome. Thus I will start by making one consensus sequence for the crypton with the most hits in each species

### 18-10-19
extract more flanks, get the consensus' done, make phyla tree (raxml) on amino acid level, get some crypton H sequences, include V's and, two or three from every other group. dotplots and maybe a phylogeny of the species that we found new ones in. Make a schematic figure of a well annotated crypton V. See if I can find a metazoan tree to use for the "we found em here" 

